Get all contracts

Smightym8 opened the pull request:
This pull request introduces several new features and improvements related to contract management in the `SMAIAXBackend` project. The primary changes include adding a new service for listing contracts, creating a new endpoint for fetching contracts, and updating DTOs and exception handling.

### New Features:

* [`src/SMAIAXBackend.API/ApplicationConfigurations/ServiceExtensions.cs`](diffhunk://#diff-93cea4da8c7048a377ce94f460260ad95c6a26fab95e5945d32754acec479428R27): Added `IContractListService` to the service configurations.
* [`src/SMAIAXBackend.API/Endpoints/Contract/ContractEndpoints.cs`](diffhunk://#diff-5b3b0136e990fe48cd4ea656f0564ab7e822fa36ae57bb7e9de1c56d83209431R22-R28): Added a new GET endpoint to fetch a list of contracts.
* [`src/SMAIAXBackend.API/Endpoints/Contract/GetContractsEndpoint.cs`](diffhunk://#diff-17b08b49dda1685546f8eb74d06594f7e2051efa5841f15893dc75047cec7455R1-R16): Implemented the handler for the new GET endpoint to fetch contracts.

### DTO and Exception Handling:

* [`src/SMAIAXBackend.Application/DTOs/ContractOverviewDto.cs`](diffhunk://#diff-05ffb00d294b19a392ac5964f113b202b3fe29245aca8572d4d844661f4134c1R1-R26): Created `ContractOverviewDto` to represent contract data.
* [`src/SMAIAXBackend.Application/Exceptions/TenantNotFoundException.cs`](diffhunk://#diff-c7d329732d1407393ba552bd3e441b32c7acfa71a839ef01e0ad60436c549776L3-R13): Updated `TenantNotFoundException` to include a constructor that accepts only `tenantId`.

### Service Implementations:

* [`src/SMAIAXBackend.Application/Services/Implementations/ContractListService.cs`](diffhunk://#diff-9f73ca102df0b65d83101cbc450096502c91431ccba92dd7f2f715d6517a8ff1R1-R48): Implemented `ContractListService` to handle the logic for fetching contracts.
* [`src/SMAIAXBackend.Application/Services/Interfaces/IContractListService.cs`](diffhunk://#diff-84c621147052edc7f833750e56e5ccb43f3ec64780dd9add44d065034d3b1689R1-R8): Defined the interface `IContractListService` with a method to get contract lists.

### Repository Updates:

* [`src/SMAIAXBackend.Domain/Repositories/IContractRepository.cs`](diffhunk://#diff-fd079a5f2c777038465f7ca20cb356776849559d00747861b70e02540b42108dR10): Added a method to fetch contracts for a tenant.
* [`src/SMAIAXBackend.Infrastructure/Repositories/ContractRepository.cs`](diffhunk://#diff-d9549cead4c43aec31512052b0ea734163127fe5af075eada96ca01a622ce520R22-R28): Implemented the method to fetch contracts for a tenant in the `ContractRepository`.

Smightym8 committed:
feat(contract): implement get all contracts

- Created endpoint to get all contracts
- Implemented ContractListService to get all contracts where the current user is the buyer

1r-1na commented on src/SMAIAXBackend.Application/DTOs/ContractOverviewDto.cs:
```diff
@@ -0,0 +1,26 @@
+{
+    [Required] public Guid Id { get; set; } = id;
+    [Required] public DateTime CreatedAt { get; set; } = createdAt;
+    [Required] public string PolicyName { get; set; } = policyName;
```
Why do we add the policy fields separately and do not just add PolicyDto?
If anything will be adjusted in PolicyDto, we have to manage it here too.

Smightym8 committed:
refactor(contract): update ContractOverviewDto to include PolicyDto

- Replaced policy properties from ContractOverviewDto with PolicyDto
- Updated FromContract method to take measurementCount as parameter and to create a PolicyDto from the provided policy
- Refactored GetContractsAsync to get the measurement count for the policy.
